tests/test_singleecujob.py

```diff
@@ -102,45 +129,24 @@ class Context(NamedTuple):
                     revision="0.12.34",
                     entrypoint="CalledClass",
                     library_refs=[
-                        "my.favourite.lib"
+                        OdxLinkRef("my.favourite.lib", doc_frags)
                     ]
                 )
             ],
-            input_params=[
-                InputParam(
-                    short_name="inputParam",
-                    physical_default_value="Yes!",
-                    dop_base_ref=self.context.inputDOP.id
-                )
-            ],
-            output_params=[
-                OutputParam(
-                    id="ID.outputParam",
-                    semantic="DATA",
-                    short_name="outputParam",
-                    long_name="The Output Param",
-                    description="<p>The one and only output of this job.</p>",
-                    dop_base_ref=self.context.outputDOP.id
-                )
-            ],
-            neg_output_params=[
-                NegOutputParam(
-                    short_name="NegativeOutputParam",
-                    description="<p>The one and only output of this job.</p>",
-                    dop_base_ref=self.context.negOutputDOP.id
-                )
-            ]
+            input_params=input_params,
+            output_params=output_params,
+            neg_output_params=neg_output_params,
         )
 
         self.singleecujob_odx = f"""
-            <SINGLE-ECU-JOB ID="{self.singleecujob_object.id}">
+            <SINGLE-ECU-JOB ID="{self.singleecujob_object.id.local_id}">
```

Comment:
Maybe a helper that takes OdxLinkId as input and return all attributes?

Reply:
I guess that won't make things clearer: The problem is that for IDs, we don't want the document fragment stuff, since this can only be specified for references, but not for IDs.